rotkehlchen/api/v1/resources.py

```diff
@@ -2370,26 +2371,28 @@ def post(self, file: FileStorage) -> Response:
         return response
 
 
-class DBSnapshotExportingResource(BaseMethodView):
-    post_schema = SnapshotExportingSchema()
+class DBSnapshotsResource(BaseMethodView):
+    get_schema = SnapshotQuerySchema()
+    upload_schema = SnapshotImportingSchema()
+    patch_schema = SnapshotEditingSchema()
+    delete_schema = SnapshotTimestampQuerySchema()
 
     @require_loggedin_user()
-    @use_kwargs(post_schema, location='json')
-    def post(self, timestamp: Timestamp, path: Path) -> Response:
-        return self.rest_api.export_user_db_snapshot(timestamp=timestamp, path=path)
-
+    @use_kwargs(get_schema, location='json_and_query_and_view_args')
+    def get(
+            self,
+            action: Optional[str],
+            timestamp: Timestamp,
+            path: Optional[Path],
+    ) -> Response:
+        if action == 'export':
+            #
+            if path is not None:
```

Comment:
Because `GET` is the only verb left. I used it exporting of snapshot.

- GET with `?action=download` is for downloading a snapshot
- GET with `?action=export&path=/home/xyz` is for exporting a snapshot
- GET without `action` query param is for returning the snapshot as json response

Reply:
Hmm this is quite a weird behaviour. Needs to be documented properly AND communicated to the frontend properly to make sure it's used correctly.